src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.cs

```diff
@@ -22,6 +22,23 @@ private void ValidatePinNumber(int pinNumber)
             }
         }
 
+        private void ValidatePinIsInput(int pinNumber)
+        {
+            ValidatePinNumber(pinNumber);
+            if (!IsPinOpen(pinNumber))

```

Comment:
Seems like this method will only be called when unregistering a pin callback, which is already ensuring that the pin was opened previously, so this check might just be doing it twice.

Reply:
See above.